src/rgw/rgw_amqp.cc

```diff
@@ -135,7 +137,10 @@ struct connection_t {
     reply_type(AMQP_RESPONSE_NORMAL),
     reply_code(RGW_AMQP_NO_REPLY_CODE),
     ref_count(0),
-    cct(nullptr) {}
+    cct(nullptr),
+    next_reconnect(std::chrono::system_clock::now()),
```

Comment:
ceph has its own clocks in `common/ceph_time.h`, please use these:
```diff
-  std::chrono::system_clock::time_point next_reconnect;
+  ceph::real_time next_reconnect;
 
-    next_reconnect(std::chrono::system_clock::now()),
+    next_reconnect(ceph::real_clock::now()),
 
-  const std::chrono::system_clock::duration idle_time;
+  const ceph::timespan idle_time;
```

there's also a lower-resolution `ceph::coarse_real_clock` that's cheaper to sample if that fits your use case